src/org/thoughtcrime/securesms/ConversationListActivity.java

```diff
@@ -128,6 +130,63 @@ protected void onCreate(Bundle icicle, boolean ready) {
     selfAvatar.setOnClickListener(v -> AccountManager.getInstance().showSwitchAccountMenu(this));
 
     refresh();
+
+    if (BuildConfig.DEBUG) checkNdkArchitecture();
+  }
+
+  /**
+   * If the build script is invoked with a specific architecture (e.g.`./ndk-make.sh arm64-v8a`), it
+   * will compile the core only for this arch. This method checks if the arch was correct.
+   *
+   * In order to do this, `ndk-make.sh` writes its argument into the file `ndkArch`.
+   * `getNdkArch()` in `build.gradle` then reads this file and its content is assigned to
+   * `BuildConfig.NDK_ARCH`.
+   */
+  @SuppressWarnings("ConstantConditions")
+  private void checkNdkArchitecture() {
+    boolean wrongArch = false;
+
+    if (!TextUtils.isEmpty(BuildConfig.NDK_ARCH)) {
+      String archProperty = System.getProperty("os.arch");
+      String arch;
+
+      if (archProperty.startsWith("armv7")) arch = "armeabi-v7a";
+      else if (archProperty.equals("aarch64")) arch = "arm64-v8a";
+      else if (archProperty.equals("i686")) arch = "x86";
+      else if (archProperty.equals("x86_64")) arch = "x86_64";
+      else {
+        Log.e(TAG, "Unknown os.arch: " + archProperty);
+        arch = "";
+      }
+
+      if (!arch.equals(BuildConfig.NDK_ARCH)) {
+        wrongArch = true;
+
+        String message;
+        if (arch.equals("")) {
+          message = "This phone has the unknown architecture " + archProperty + ".\n\n"+
+                  "Please open an issue at https://github.com/deltachat/deltachat-android/issues.";
+        } else {
+          message = "Apparently you used `ndk-make.sh " + BuildConfig.NDK_ARCH + "`, but this device is " + arch + ".\n\n" +
+                  "You can use the app, but changes you made to the Rust code were not applied.\n\n" +
+                  "To compile in your changes, you can:\n" +
+                  "- Either run `ndk-make.sh " + arch + "` to build only for " + arch + " in debug mode\n" +
+                  "- Or run `ndk-make.sh` without argument to build for all architectures in release mode\n\n" +
+                  "If something doesn't work, please open an issue at https://github.com/deltachat/deltachat-android/issues!!";
```

Comment:
The text is not linkified - "ndk-make.sh" was also linkified when I enabled linkifying, and it's probably not woth it as most developers will not open an issue from their phone, anyway I guess.